### PR TITLE
[ts] mesh layers

### PR DESCRIPTION
**Pull request by Pessimistress**
For #5589 

Waiting on https://github.com/visgl/luma.gl/pull/1647

#### Change List
- Convert `ScenegraphLayer` and `SimpleMeshLayer` to TS
- Enable TS compilation for the mesh-layers module

@zbigg After this is merged you should be able to uncomment the geo-layers and carto modules in tsconfig.build.json. There seems to be some minor errors so I did not include it in this PR.

### CODE REVIEW

File: modules/core/src/lib/attribute/data-column.ts [hunk 1]
```diff
@@ -84,8 +84,9 @@ function resolveDoublePrecisionShaderAttributes(
 }
 
 export type DataColumnOptions<Options> = Options &
-  BufferAccessor & {
+  Omit<BufferAccessor, 'size'> & {
```
Comment (felixpalmer) [inline modules/core/src/lib/attribute/data-column.ts:87]: Would it not be simpler to just relax `BufferAccessor.size` to `number`?

File: modules/mesh-layers/src/scenegraph-layer/scenegraph-layer.ts [hunk 2]
```diff
@@ -31,8 +37,107 @@ import {MATRIX_ATTRIBUTES, shouldComposeModelMatrix} from '../utils/matrix';
...
+} from '@deck.gl/core';
+
 const DEFAULT_COLOR = [255, 255, 255, 255];
 
+export type ScenegraphLayerProps<DataT = any> = _ScenegraphLayerProps<DataT> & LayerProps<DataT>;
+
+type _ScenegraphLayerProps<DataT> = {
+  // TODO - define in luma.gl
+  /**
+   * A url to glTF model or scenegraph loaded via a [scenegraph loader](https://loaders.gl/docs/specifications/category-scenegraph)
```
Comment (felixpalmer) [inline modules/mesh-layers/src/scenegraph-layer/scenegraph-layer.ts:56]:
```suggestion
   * A url for a glTF model or scenegraph loaded via a [scenegraph loader](https://loaders.gl/docs/specifications/category-scenegraph)
```

File: modules/mesh-layers/src/scenegraph-layer/scenegraph-layer.ts [hunk 3]
```diff
@@ -102,25 +219,26 @@ export default class ScenegraphLayer extends Layer {
...
 
-  finalizeState() {
-    super.finalizeState();
+  finalizeState(context: LayerContext) {
+    super.finalizeState(context);
     this._deleteScenegraph();
   }
 
-  _updateScenegraph(props) {
+  private _updateScenegraph() {
```
Comment (felixpalmer) [inline modules/mesh-layers/src/scenegraph-layer/scenegraph-layer.ts:238]:
```suggestion
  private _updateScenegraph(): void {
```

File: modules/mesh-layers/src/simple-mesh-layer/simple-mesh-layer.ts [hunk 4]
```diff
@@ -32,7 +40,11 @@ import {MATRIX_ATTRIBUTES, shouldComposeModelMatrix} from '../utils/matrix';
 import vs from './simple-mesh-layer-vertex.glsl';
 import fs from './simple-mesh-layer-fragment.glsl';
 
-function validateGeometryAttributes(attributes, useMeshColors) {
+import type {LayerProps, UpdateParameters, Accessor, Position, Color, Texture} from '@deck.gl/core';
+import type {MeshAttribute, MeshAttributes} from '@loaders.gl/schema';
+import type {Geometry as GeometryType} from '@luma.gl/engine';
+
+function validateGeometryAttributes(attributes: Record<string, any>, useMeshColors: boolean) {
```
Comment (felixpalmer) [inline modules/mesh-layers/src/simple-mesh-layer/simple-mesh-layer.ts:47]:
```suggestion
function validateGeometryAttributes(attributes: Record<string, any>, useMeshColors: boolean): void {
```

File: yarn.lock [hunk 5]
```diff
@@ -2403,65 +2403,65 @@
   dependencies:
     "@babel/runtime" "^7.3.1"
 
-"@luma.gl/constants@8.5.12", "@luma.gl/constants@^8.5.12":
-  version "8.5.12"
-  resolved "https://registry.yarnpkg.com/@luma.gl/constants/-/constants-8.5.12.tgz#4fe916634286f8ad9f41c0e688d75db904b6ea60"
-  integrity sha512-XKW4/Ur6azfoytV7R+84KG5h4E+FwqVOnxAAYyXk8kbNoI6uY829bj+/TxuUV6oJCzflmSCjs1Z4tVbsCVn5Lg==
+"@luma.gl/constants@8.5.13", "@luma.gl/constants@^8.5.13":
```
Comment (felixpalmer) [inline yarn.lock:2406]: The root `package.json` has `@luma.gl/test-utils": "^8.5.12"`, worth bumping to 8.5.13 also?

**Review by felixpalmer — APPROVED** [review]

**Review by zbigg — APPROVED** [review]

**Pull request merged**